### ISSUE TITLE
Bug: tree traversal does not produce minimal set of chunks.

### ISSUE THREAD

**Issue by rklaehn** [n0-computer/bao-tree#41]
E.g. when encoding the range chunk 16..chunk 24 of a file of 32 chunks, it should return a single leaf but returns multiple.

See failing test in https://github.com/n0-computer/bao-tree/tree/sub-chunk-group-bug

This will only affect you if you ask for sub chunk groups. But it is still bad. Changing this will require an ALPN increase for iroh-bytes.

**Comment by rklaehn** [comment]
Maybe this is just misuse? The split fn at least seems to try to do the right thing and canonicalize the ranges. In any case, this should be fixable by fixing split to properly canonicalize ranges to all.

```rust
pub(crate) fn split(
    ranges: &RangeSetRef<ChunkNum>,
    mid: ChunkNum,
) -> (&RangeSetRef<ChunkNum>, &RangeSetRef<ChunkNum>) {
    let (a, mut b) = ranges.split(mid);
    // check that a does not contain a redundant boundary at or after mid
    debug_assert!(a.boundaries().last() < Some(&mid));
    // Replace b with the canonicalized version if it starts at or before mid.
    // This is necessary to be able to check it with RangeSetRef::is_all()
    if b.boundaries().len() == 1 && b.boundaries()[0] <= mid {
        b = RangeSetRef::new(&[ChunkNum(0)]).unwrap();
    }
    (a, b)
}
```

**Comment by rklaehn** [comment]
Fixed by #43
